modules/monitoring-creating-cluster-monitoring-configmap.adoc

```diff
@@ -44,3 +32,9 @@ metadata:
 data:
   config.yaml: |
 ----
+
+. Start editing the `cluster-monitoring-config` ConfigMap:
```

Comment:
For this step I think it would be useful to clarify why it is we are asking the reader to edit the ConfigFile if it already exists. I suggest something like:

```
. If the ConfigMap already exists:
.. editing the `cluster-monitoring-config` ConfigMap:
+
[source,terminal]
----
$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
----
+
.. Then add the `data` section if it does not already exist:
+
[source,yaml]
----
apiVersion: v1
kind: ConfigMap
metadata:
  name: cluster-monitoring-config
  namespace: openshift-monitoring
data:
  config.yaml: |
----
```

Reply:
I'll double check about this one. This section is "creating configmap" and it is followed by "configuring" section: https://docs.openshift.com/container-platform/4.4/monitoring/cluster_monitoring/configuring-the-monitoring-stack.html#configuring-the-cluster-monitoring-stack_configuring-monitoring
it feels like too much duplication

Reply:
Thanks for highlighting that. I think that the two sections should remain separate, because the module on Enabling monitoring of your own services [1] references the Creating cluster monitoring ConfigMap section as a starting point.

I am also working on a full restructuring of the whole book for 4.6 in the PR I linked in my other comment, as we introduce full support for user-defined workload monitoring.

Reply:
I see what you mean about it feeling like too much duplication actually. Another option would be to update the 'Enabling monitoring of your own services' section for 4.5 and earlier, so that it looks more like the one I have in my 4.6 PR [1], which adds an explicit step to the procedure to add a ConfigMap if it does not exist. That way we do not need to refer to the 'Creating cluster monitoring ConfigMap' module from it.

[1] https://github.com/openshift/openshift-docs/pull/24722/files#diff-0d6431e3992282fc3cfa5fadb9500d2d

Reply:
That would also help with https://bugzilla.redhat.com/show_bug.cgi?id=1865805.

Reply:
@pneedle-rh this is great! I think this: https://github.com/openshift/openshift-docs/pull/24722/files#diff-89302581129a792feee2baace787a6dbR15-R42
makes my PR obsolete. Do you agree that I close my PR and let you clean drive yours without the need to duplicate the work and handle merge conflicts.

Reply:
@olga-mir that PR is for 4.6 onward. I am not restructuring 4.5 and so your PR is still relevant for 4.5 and earlier, I would say. I will then handle any conflicts for 4.6.

Reply:
@olga-mir on advice from one of our experienced team members, I have updated the procedure in my PR to use bullet points instead of numbering as with the following example:

```
** If the ConfigMap does not exist:
.. Create the following YAML manifest. In this example the file is called `cluster-monitoring-config.yaml:
```
Please see https://github.com/openshift/openshift-docs/pull/24722/files#diff-89302581129a792feee2baace787a6db for more details.

Reply:
Thank you. I have updated this PR. Can you explain what do you mean that your PR is targeting 4.6 while mine is 4.5? they are both against master and I don't see any versions in filenames or ifdefs.

Reply:
@olga-mir thank you for your response. The PR I am working on includes an extensive restructuring of the Monitoring book for 4.6 and later. It also includes a lot of new content for the user workload monitoring feature that is being introduced in 4.6. When it is approved, I will merge it into master and cherry pick it into the enterprise-4.6 branch. But, the changes will not be cherry picked into 4.3, 4.4. or 4.5.

I assumed that your PR will be merged into master and then cherry picked into 4.3, 4.4 and 4.5, but I am not sure because there are no labels in relation to that yet.

I think it is also likely that your changes will be merged into master before mine, because I still have a lot to do and there will be a detailed review that might take us up to the end of September. So, this will likely cause a merge conflict in my feature branch when I fetch and rebase to bring in your (then merged) changes from the master branch. At that point I will make sure that my PR reflects what you have done. Then when I merge my PR to master and CP to 4.6, it will reflect both of our updates.

Please let me know if you have any further questions on this.

Reply:
Thank you for detailed explanation!